src/containers/tutor-home-page/add-photo-step/AddPhotoStep.jsx

```diff
@@ -1,11 +1,101 @@
-import { Box } from '@mui/material'
-import { style } from '~/containers/tutor-home-page/add-photo-step/AddPhotoStep.style'
+import { Box, Button, Typography } from '@mui/material'
+import { useState, useEffect } from 'react'
+import useFileReader from '~/hooks/use-file-reader'
+import { validationData, clearButtonNameMaxLength } from './constants'
+import { useTranslation } from 'react-i18next'
+import CloseIcon from '@mui/icons-material/Close'
+import useBreakpoints from '~/hooks/use-breakpoints'
+import DragAndDrop from '~/components/drag-and-drop/DragAndDrop'
+import FileUploader from '~/components/file-uploader/FileUploader'
+import {
+  style,
+  dragStyles,
+  dragStylesWithBorder
+} from '~/containers/tutor-home-page/add-photo-step/AddPhotoStep.style'
+
+const AddPhotoStep = ({ btnsBox, onFileUpload = () => {} }) => {
+  const [file, setFile] = useState(null)
+  const { fileDataURL, readFileAsDataURL, resetFileDataURL } = useFileReader()
+  const [fileName, setFileName] = useState(null)
+  const { isMobile, isTablet } = useBreakpoints()
+  const { t } = useTranslation()
+
+  const emitter = ({ files }) => {
+    const firstfile = files[0]
+    setFile(firstfile)
+    onFileUpload(firstfile)
+    setFileName(getShortenedFileName(firstfile.name, clearButtonNameMaxLength))
+    readFileAsDataURL(firstfile)
+  }
+
+  const handleClear = () => {
+    setFile(null)
+    resetFileDataURL()
+    setFileName(null)
+  }
+
+  useEffect(() => {
+    onFileUpload(file)
+  }, [file])
```

Comment:
Pay attention to the warning from the linter here.

Reply:
Thanks for the comment, reviewed this block and realized that I can remove the extra useState and useEffect since this function is also called in the emitter function and gets the file.